awx_collection/plugins/module_utils/tower_api.py

```diff
@@ -531,8 +531,11 @@ def update_if_needed(self, existing_item, new_item, handle_response=True, on_upd
             # If we have an item, we can see if it needs an update
             try:
                 item_url = existing_item['url']
-                item_name = existing_item['name']
-                item_type = existing_item['url']
+                item_type = existing_item['type']
+                if item_type == 'user':
+                    item_name = existing_item['username']
+                else:
+                    item_name = existing_item['name']
```

Comment:
This change addresses the issue filed here: https://github.com/ansible/awx/issues/6128

Reply:
https://github.com/ansible/awx/issues/6128 also affects roles, I'm not sure this will fix the `tower_role` part of it

Reply:
oh looks like we havent converted role, nevermind